torch/_inductor/codegen/cuda/cuda_scheduling.py

```diff
@@ -1,10 +1,80 @@
+from typing import cast, List
+
 from ... import config
 from ...codecache import code_hash, get_path
+from ...scheduler import FusedSchedulerNode, Scheduler, SchedulerNode
 from ...utils import get_fused_kernel_name, get_kernel_metadata
 from ...virtualized import V
-
 from ..common import IndentedBuffer
 from ..triton import TritonScheduling
+from .cuda_kernel import CUDATemplateBuffer
+
+
+class CUDASchedulerNode(SchedulerNode):
+    def __init__(self, scheduler: Scheduler, node: CUDATemplateBuffer, group_fn):
+        assert isinstance(node, CUDATemplateBuffer)
+        super().__init__(scheduler, node, group_fn)
+        self.group_fn = group_fn  # keeping this to enable cloning during fuse_epilogue
+
+    def can_fuse_epilogue(self, other_node: SchedulerNode) -> bool:
```

Comment:
These two methods mostly delegate to CUDATemplateBuffer, which has the necessary information.